### PR TITLE
Login page with Unit test cases

### PR DESCRIPTION
**Pull request by kboodi**
Created login page with two tag switches
Login with email:
image

Login with mobile:
image

Form validations:
image

image

image

image

Login failure message:
image

Issue link:
https://abhirams.atlassian.net/browse/P3ECWFC-55

### CODE REVIEW

File: src/pages/Login/Login.scss [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  }
+  a.nav-link:hover {
+    color: $black !important;
+    font-weight: 600;
+    background-color: lightgrey;
+  }
+  a.nav-link.active {
+    color: $black !important;
+    font-weight: 600;
+    // background-color: lightgrey;
```
Comment (pashedin) [inline src/pages/Login/Login.scss]: please remove any and all unnecessary comments from the PR

File: src/pages/Login/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+import { Link, useNavigate } from "react-router-dom";
+import { useForm } from "react-hook-form";
+import { contactpattern, emailpattern } from "../../constants";
+import { useDispatch, useSelector } from "react-redux";
+import { login } from "../../redux/store/Authentication/actions";
+import { LoginStatus } from "../../models/LoginStatus";
+
+function Login() {
+  const [tabState, SetTabState] = useState("with-email");
+  const [userMainField, setuserMainField] = useState("Email address");
```
Comment (pashedin) [inline src/pages/Login/Login.tsx]: please maintain camelCase `SetTabState` should be `setTabState` similarly should have been `setUserMainField`

Comment (pashedin) [reply src/pages/Login/Login.tsx]: Also we do not need to maintain 2 different state variables what we have here can be achieved using either of these variables

File: src/pages/Login/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const isAuthenticated = useSelector(
+    (state: any) => state.authentiaction.authenticated
+  );
+  const [displayMsg, setDisplayMsg] = useState(false);
```
Comment (pashedin) [inline src/pages/Login/Login.tsx]: displayMsg is saving either True and false. in that case please rethink the variable name.

File: src/pages/Login/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+                  minLength: {
+                    value: 10,
+                    message: "Mobile number cannot be less than 10 digits",
+                  },
+                  maxLength: {
+                    value: 10,
+                    message: "Mobile number cannot be more than 10 digits",
+                  },
+                  pattern: {
+                    value: /^[6-9]\d{9}$/,
```
Comment (pashedin) [inline src/pages/Login/Login.tsx]: add/use from constants file

File: src/redux/store/Authentication/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+const initialState: { authenticated: LoginStatus } = {
+  authenticated: LoginStatus.LoggedOut,
+};
+
+const AuthenticationReducer = (state = initialState, action: any) => {
+  switch (action.type) {
+    case LOGIN:
+      const { loginCredentials } = action;
+      const authenticated =
+        loginCredentials?.password === "L#(qc{f}TaJu4%4k"
```
Comment (pashedin) [inline src/redux/store/Authentication/reducer.ts:13]: I would like to have a simpler password for Devs to use, please.

**Review by pashedin — DISMISSED** [review]

File: src/pages/Login/Login.tsx (hunk 2 above)
Comment (kboodi) [reply src/pages/Login/Login.tsx]: updated name with camelCase and removed the unnecessary state used (userMainField)

File: src/pages/Login/Login.scss (hunk 1 above)
Comment (kboodi) [reply src/pages/Login/Login.scss]: removed unnecessary comments

File: src/pages/Login/Login.tsx (hunk 3 above)
Comment (kboodi) [reply src/pages/Login/Login.tsx]: Changed the variable with appropriate name

File: src/redux/store/Authentication/reducer.ts (hunk 5 above)
Comment (kboodi) [reply src/redux/store/Authentication/reducer.ts:13]: Not changed as discussed over the call

File: src/pages/Login/Login.tsx (hunk 4 above)
Comment (kboodi) [reply src/pages/Login/Login.tsx]: used constants.ts global file

**Review by pant-manas — APPROVED** [review]

**Pull request merged**